### PR TITLE
Send an EOF from charts.d.plugin before exit

### LINKED ISSUE #14531: [Bug]: Problem with charts.d.plugin restarts since 1.38.0
### Bug description

I run [6 instances](https://learn.netdata.cloud/docs/agent/collectors/charts.d.plugin#running-multiple-instances-of-chartsdplugin) of charts.d.plugin. By default they restart every 4 hours (`restart_timeout=$[3600 * 4]`). After this period some of them stops working. The problem is that if Netdata restarts multiple instances at the same time it "forgets" to start some of them again.

For this test I set restart period to 1 minute. In this case `charts-mab-mysql.d` plugin stopped working.
```
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-ethtool.d] : thread created with task id 51973
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-ethtool.d] : set name of thread 51973 to PD[charts-mab-e
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-switch.d] : thread created with task id 51972
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-switch.d] : set name of thread 51972 to PD[charts-mab-s
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-mysql.d] : thread created with task id 51976
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-mysql.d] : set name of thread 51976 to PD[charts-mab-m
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-iptables.d] : thread created with task id 51978
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-msa.d] : thread created with task id 51980
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-msa.d] : set name of thread 51980 to PD[charts-mab-m
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-iptables.d] : set name of thread 51978 to PD[charts-mab-i
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab.d] : thread created with task id 51984
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab.d] : set name of thread 51984 to PD[charts-mab.d
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-switch.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-switch.d.plugin' running on pid 51999
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-ethtool.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-ethtool.d.plugin' running on pid 52000
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-mysql.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-mysql.d.plugin' running on pid 52001
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab.d.plugin' running on pid 52005
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-msa.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-msa.d.plugin' running on pid 52267
2023-02-15 07:54:02: netdata INFO  : PD[charts-mab-iptables.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-iptables.d.plugin' running on pid 52268

--- after 1 minute

2023-02-15 07:55:04: netdata ERROR : PD[charts-mab-iptables.d] : '/usr/libexec/netdata/plugins.d/charts-mab-iptables.d.plugin' (pid 52268) disconnected after 1525 successful data collections (ENDs).
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab.d] : read failed: end of file (errno 9, Bad file descriptor)
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab.d] : '/usr/libexec/netdata/plugins.d/charts-mab.d.plugin' (pid 52005) disconnected after 488 successful data collections (ENDs).
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab-ethtool.d] : read failed: end of file (errno 9, Bad file descriptor)
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab-ethtool.d] : '/usr/libexec/netdata/plugins.d/charts-mab-ethtool.d.plugin' (pid 52000) disconnected after 71431 successful data collections (ENDs).
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab-msa.d] : read failed: end of file (errno 9, Bad file descriptor)
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab-msa.d] : '/usr/libexec/netdata/plugins.d/charts-mab-msa.d.plugin' (pid 52267) disconnected after 3600 successful data collections (ENDs).
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab-switch.d] : read failed: end of file (errno 9, Bad file descriptor)
2023-02-15 07:55:04: netdata ERROR : PD[charts-mab-switch.d] : '/usr/libexec/netdata/plugins.d/charts-mab-switch.d.plugin' (pid 51999) disconnected after 4209 successful data collections (ENDs).
2023-02-15 07:55:05: netdata INFO  : PD[charts-mab-iptables.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-iptables.d.plugin' running on pid 9462
2023-02-15 07:55:05: netdata INFO  : PD[charts-mab.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab.d.plugin' running on pid 9466
2023-02-15 07:55:05: netdata INFO  : PD[charts-mab-ethtool.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-ethtool.d.plugin' running on pid 9471
2023-02-15 07:55:05: netdata INFO  : PD[charts-mab-msa.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-msa.d.plugin' running on pid 9474
2023-02-15 07:55:05: netdata INFO  : PD[charts-mab-switch.d] : connected to '/usr/libexec/netdata/plugins.d/charts-mab-switch.d.plugin' running on pid 9479
```

It looks like that this issue is gone if I set different `restart_timeout` for each of them. So I have a workaround for now.

### Expected behavior

Successfully restart all charts.d plugins at once.

### Steps to reproduce

1. Set [multiple charts.d.plugin instances](https://learn.netdata.cloud/docs/agent/collectors/charts.d.plugin#running-multiple-instances-of-chartsdplugin)
2. Set `restart_timeout=$[60]` for each of them.
3. After 1 minute some of them may stop working.

### Installation method

kickstart.sh

### System info

```shell
Linux server7 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=18.04
/etc/lsb-release:DISTRIB_CODENAME=bionic
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 18.04.6 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="18.04.6 LTS (Bionic Beaver)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 18.04.6 LTS"
/etc/os-release:VERSION_ID="18.04"
/etc/os-release:VERSION_CODENAME=bionic
/etc/os-release:UBUNTU_CODENAME=bionic
```

### Netdata build info

```shell
Version: netdata v1.38.1
Configure options:  '--prefix=/usr' '--sysconfdir=/etc' '--localstatedir=/var' '--libexecdir=/usr/libexec' '--libdir=/usr/lib' '--with-zlib' '--with-math' '--with-user=netdata' '--disable-cloud' '--with-bundled-protobuf' 'CFLAGS=-O2 -pipe' 'LDFLAGS='
Install type: kickstart-build
Features:
    dbengine:                   YES
    Native HTTPS:               YES
    Netdata Cloud:              NO (by user request)
    ACLK:                       NO
    TLS Host Verification:      YES
    Machine Learning:           YES
    Stream Compression:         NO
Libraries:
    protobuf:                YES (bundled)
    jemalloc:                NO
    JSON-C:                  YES
    libcap:                  NO
    libcrypto:               YES
    libm:                    YES
    tcalloc:                 NO
    zlib:                    YES
Plugins:
    apps:                    YES
    cgroup Network Tracking: YES
    CUPS:                    NO
    EBPF:                    YES
    IPMI:                    NO
    NFACCT:                  NO
    perf:                    YES
    slabinfo:                YES
    Xen:                     NO
    Xen VBD Error Tracking:  NO
Exporters:
    AWS Kinesis:             NO
    GCP PubSub:              NO
    MongoDB:                 NO
    Prometheus Remote Write: NO
Debug/Developer Features:
    Trace Allocations:       NO
```

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

Fixes #14531 

This PR adds a last will EOF when a charts.d.plugin script exits.

It appears that under some cases, `fgets` in parser will trigger at the exact moment the script is asked to exit, cleans up `/tmp` and exits. `fgets` then will wait forever. The EOF here just makes sure `fgets` will receive something before the script exits.

The issue is more apparent when there are a lot of different charts.d.plugin scripts that are restarted at the same time.

##### Test Plan

Many many thanks to @vobruba-martin for helping with this!!

Not really easy to reproduce. We've tested this on @vobruba-martin's setup and it seems to fix the issue. I don't expect it to cause any side effects, but for testing, e.g. the example charts.d.plugin, make sure it restart correctly.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by stelfrag — APPROVED** [review]

File: collectors/charts.d.plugin/charts.d.plugin.in [hunk 1]
```diff
@@ -32,6 +32,7 @@ chartsd_cleanup() {
 		[ $debug -eq 1 ] && echo >&2 "$PROGRAM_NAME: cleaning up temporary directory $TMP_DIR ..."
 		rm -rf "$TMP_DIR"
 	fi
+	echo ""
```
[the remark this answers is not in the document]
Comment (MrZammler) [reply collectors/charts.d.plugin/charts.d.plugin.in]: You're right, it'll be better to have a proper keyword...

**Comment by MrZammler** [comment]
Tested also by @vobruba-martin, thank you!

**Review by ilyam8 — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
PR is working as expected with other plugin as well:

```sh
/var/log/netdata/error.log:2023-03-08 13:54:34: netdata INFO  : PD[myfirst] : (1906@collectors/plugins.d:pluginsd_exit  ): PLUGINSD: plugin called EXIT
```

LGTM!

**Comment by vobruba-martin** [comment]
Side note: I suspect the issue happens with other plugins as well. But it is not as visible to me as with the charts.d plugin. During today's tests I saw `cgroups`, `go.d` and `timex` plugins stalled when I tried to restart Netdata.

**Comment by ilyam8** [comment]
@vobruba-martin that is not possible, if there is an issue then it is a different issue 100% (cgroups and timex are internal plugins)

**Comment by vobruba-martin** [comment]
@ilyam8 I'm sorry, I wanted to add the comment to the related issue and not to this PR.

**Pull request merged**